### PR TITLE
file logging is now disabled by default unless the OSVR_FILE_LOGGING_ENABLED environment variable is defined and not set to 0

### PR DESCRIPTION
**Pull request by JeroMiya**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# VS Code
+.vscode/
+
```
Comment (mars979) [inline .gitignore]: Was this an intentional addition to the gitignore?

**Comment by godbyk** [comment]
To be clear, it's not file logging per se that is the problem. It's the asynchronous logging queue which is used by all loggers.  Specifically, [this line](https://github.com/OSVR/OSVR-Core/blob/master/src/osvr/Util/LogRegistry.cpp#L209). 

The queue size refers to the number of messages the queue can hold. Right now it's set to 64k messages. We can reduce this drastically to save memory. The value must be a power of two, though (i.e., 2^x).

File: .gitignore (hunk 1 above)
Comment (JeroMiya) [inline .gitignore]: Yes. Sorry, should have put it in another commit. .vscode is a temp directory from visual studio code.
